internal/controller/ols_app_postgres_assets.go

```diff
@@ -0,0 +1,346 @@
+	}
+	defaultPermission := int32(0600)
+	tlsCertsVolume := corev1.Volume{
+		Name: "secret-" + PostgresCertsSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName:  PostgresCertsSecretName,
+				DefaultMode: &defaultPermission,
+			},
+		},
+	}
+	bootstrapVolume := corev1.Volume{
+		Name: "secret-" + PostgresBootstrapSecretName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: PostgresBootstrapSecretName,
+			},
+		},
+	}
+	configVolume := corev1.Volume{
+		Name: PostgresConfigMap,
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{Name: PostgresConfigMap},
+			},
+		},
+	}
+	dataVolume := corev1.Volume{
+		Name: PostgresDataVolume,
+		VolumeSource: corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		},
+	}
+	volumes := []corev1.Volume{tlsCertsVolume, bootstrapVolume, configVolume, dataVolume, getPostgresCAConfigVolume()}
+	postgresTLSVolumeMount := corev1.VolumeMount{
+		Name:      "secret-" + PostgresCertsSecretName,
+		MountPath: OLSAppCertsMountRoot,
+		ReadOnly:  true,
+	}
+	bootstrapVolumeMount := corev1.VolumeMount{
+		Name:      "secret-" + PostgresBootstrapSecretName,
+		MountPath: PostgresBootstrapVolumeMount,
+		SubPath:   PostgresExtensionScript,
+		ReadOnly:  true,
+	}
+	configVolumeMount := corev1.VolumeMount{
+		Name:      PostgresConfigMap,
+		MountPath: PostgresConfigVolumeMount,
+		SubPath:   PostgresConfig,
+	}
+	dataVolumeMount := corev1.VolumeMount{
+		Name:      PostgresDataVolume,
+		MountPath: PostgresDataVolumeMount,
+	}
+	volumeMounts := []corev1.VolumeMount{postgresTLSVolumeMount, bootstrapVolumeMount, configVolumeMount, dataVolumeMount, getPostgresCAVolumeMount(path.Join(OLSAppCertsMountRoot, PostgresCAVolume))}
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PostgresDeploymentName,
+			Namespace: r.Options.Namespace,
+			Labels:    generatePostgresSelectorLabels(),
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &cacheReplicas,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: generatePostgresSelectorLabels(),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: generatePostgresSelectorLabels(),
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            PostgresDeploymentName,
+							Image:           r.Options.LightspeedServicePostgresImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "server",
+									ContainerPort: PostgresServicePort,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							VolumeMounts: volumeMounts,
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("500m"),
```

Comment:
half a core as a minimum request?  that seems high, how did you come up with this value?

Reply:
Ran this tool and reset values based on the below recommendations.
```
┃ Number ┃ Namespace                ┃ Name                    ┃ Pods ┃ Old Pods ┃ Type        ┃ Container                ┃ CPU Diff ┃ CPU Requests           ┃ CPU Limits        ┃ Memory Diff ┃ Memory Requests         ┃ Memory Limits     ┃
├────────┼──────────────────────────┼─────────────────────────┼──────┼──────────┼─────────────┼──────────────────────────┼──────────┼────────────────────────┼───────────────────┼─────────────┼─────────────────────────┼───────────────────┤
│    60. │ openshift-lightspeed     │ lightspeed-app-server   │ 2    │ 97       │ Deployment  │ lightspeed-service-api   │ +538m    │ (+269m) 100m -> 369m   │ 500m -> unset     │ +312Mi      │ (+156Mi) 1024Mi ->      │ 2048Mi -> 1180Mi  │
│        │                          │                         │      │          │             │                          │          │                        │                   │             │ 1180Mi                  │                   │
│    61. │                          │                         │      │          │             │ lightspeed-service-user… │ -80m     │ (-40m) 50m -> 10m      │ 100m -> unset     │ -56Mi       │ (-28Mi) 128Mi -> 100Mi  │ 256Mi -> 100Mi    │
├────────┼──────────────────────────┼─────────────────────────┼──────┼──────────┼─────────────┼──────────────────────────┼──────────┼────────────────────────┼───────────────────┼─────────────┼─────────────────────────┼───────────────────┤
│    62. │ openshift-lightspeed     │ lightspeed-console-plu… │ 2    │ 100      │ Deployment  │ lightspeed-console-plug… │ +0m      │ (+0m) 10m -> 10m       │ 100m -> unset     │ +100Mi      │ (+50Mi) 50Mi -> 100Mi   │ 100Mi -> 100Mi    │
├────────┼──────────────────────────┼─────────────────────────┼──────┼──────────┼─────────────┼──────────────────────────┼──────────┼────────────────────────┼───────────────────┼─────────────┼─────────────────────────┼───────────────────┤
│    63. │ openshift-lightspeed     │ lightspeed-postgres-se… │ 1    │ 72       │ Deployment  │ lightspeed-postgres-ser… │ -489m    │ (-489m) 500m -> 11m    │ unset             │ -248Mi      │ (-248Mi) 512Mi -> 264Mi │ unset -> 264Mi 
├────────┼──────────────────────────┼─────────────────────────┼──────┼──────────┼─────────────┼──────────────────────────┼──────────┼────────────────────────┼───────────────────┼─────────────┼─────────────────────────┼───────────────────┤
```